src/gardenlinux/github/__main__.py

```diff
@@ -0,0 +1,836 @@
+
+
+def _parse_match_section(pkg_list: list):
+    output = ""
+    for pkg in pkg_list:
+        # If is dict, the package has additional information relevant for release notes
+        if isinstance(pkg, dict):
+            pkg_string = next(iter(pkg))
+            output += f"\n{pkg_string}:\n"
+            for item in pkg[pkg_string]:
+                for k, v in item.items():
+                    output += f"  * {k}: {v}\n"
+    return output
+
+
+def release_notes_changes_section(gardenlinux_version):
+    """
+    Get list of fixed CVEs, grouped by upgraded package.
+    Note: This result is not perfect, feel free to edit the generated release notes and
+    file issues in glvd for improvement suggestions https://github.com/gardenlinux/glvd/issues
+    """
+    try:
+        url = f"https://glvd.ingress.glvd.gardnlinux.shoot.canary.k8s-hana.ondemand.com/v1/patchReleaseNotes/{gardenlinux_version}"
+        response = requests.get(url, timeout=REQUESTS_TIMEOUTS)
+        response.raise_for_status()  # Will raise an error for bad responses
+        data = response.json()
+
+        if len(data["packageList"]) == 0:
+            return ""
+
+        output = [
+            "## Changes",
+            "The following packages have been upgraded, to address the mentioned CVEs:",
+        ]
+        for package in data["packageList"]:
+            upgrade_line = (
+                f"- upgrade '{package['sourcePackageName']}' from `{package['oldVersion']}` "
+                f"to `{package['newVersion']}`"
+            )
+            output.append(upgrade_line)
+
+            if package["fixedCves"]:
+                for fixedCve in package["fixedCves"]:
+                    output.append(f"  - {fixedCve}")
+
+        return "\n".join(output) + "\n\n"
+    except:
+        # There are expected error cases, for example with versions not supported by glvd (1443.x) or when the api is not available
+        # Fail gracefully by adding the placeholder we previously used, so that the release note generation does not fail.
+        return textwrap.dedent(
+            """
+        ## Changes
+        The following packages have been upgraded, to address the mentioned CVEs:
+        **todo release facilitator: fill this in**
+        """
+        )
+
+
+def release_notes_software_components_section(package_list):
+    output = "## Software Component Versions\n"
+    output += "```"
+    output += "\n"
+    packages_regex = re.compile(
+        r"^linux-image-amd64$|^systemd$|^containerd$|^runc$|^curl$|^openssl$|^openssh-server$|^libc-bin$"
+    )
+    for entry in package_list.values():
+        if packages_regex.match(entry.deb_source):
+            output += f"{entry!r}\n"
+    output += "```"
+    output += "\n\n"
+    return output
+
+
+def release_notes_compare_package_versions_section(gardenlinux_version, package_list):
+    output = ""
+    version_components = gardenlinux_version.split(".")
+    # Assumes we always have version numbers like 1443.2
```

Comment:
Please be aware we'll introduce semantic versioning soon. This logic should use a `semver` Python package if suitable and consider the patch version to be zero if only one dot is found in the string.

Reply:
> Please be aware we'll introduce semantic versioning soon. This logic should use a `semver` Python package if suitable and consider the patch version to be zero if only one dot is found in the string.